src/main/kotlin/org/ionproject/integration/domain/calendar/BusinessObjects.kt

```diff
@@ -111,11 +106,58 @@ data class AcademicCalendar(
                     .plus("-${getTermNumber(term)}"),
                 interruptions,
                 evaluations,
-                details,
-                lectures
+                lectures,
+                otherEvents
             )
         }
 
+        private fun getLectures(
+            descriptions: List<IndexedValue<String>>,
+            dates: List<IndexedValue<String>>
+        ): List<Lectures> {
+            val lectures = mutableListOf<Lectures>()
+
+            val numberOfLecturesEvents =
+                descriptions.filter { it.value.contains(PT_LECTURES_START_REGEX.toRegex(RegexOption.IGNORE_CASE)) }
+                    .count()
+
+            /* if it's a single event then it's for all academic terms*/
+            if (numberOfLecturesEvents == 1) {
+                lectures.add(
+                    Lectures(
+                        PT_LECTURES_ALL_SECTIONS_NAME,
+                        listOf(1, 2, 3, 4, 5, 6),
+                        findFirstDateRangeFromEvent(descriptions, dates, PT_LECTURES_START_REGEX).from,
+                        findFirstDateRangeFromEvent(descriptions, dates, PT_LECTURES_END_REGEX).from,
+                    )
+                )
+                // if there are multiple events it's necessary to split per each section
+            } else {
+                // TODO: when there's different dates for different sections
+            }
+
+            return lectures
+        }
+
+        private fun findFirstDateRangeFromEvent(
+            descriptions: List<IndexedValue<String>>,
+            dates: List<IndexedValue<String>>,
+            regex: String
+        ) =
+            DateUtils.getDateRange(
+                dates[
+                    descriptions.indexOf(
+                        descriptions.find {
+                            it.value.contains(
+                                regex.toRegex(
+                                    RegexOption.IGNORE_CASE
+                                )
+                            )
+                        }
+                    )
+                ].value
```

Comment:
maybe break this into a couple of statements? a bit hard to follow

Reply:
Improvement of `findFirstDateRangeFromEvent ` for clarity